### DISCUSSION TITLE
Why use evergreen tags?

### DISCUSSION THREAD

**Discussion by KjellConnelly** [BryanJenksCommunity/FAQ#122] [Q&A] [answered]
Hi Bryan! Been watching your videos for a few months slowly when I get in a roll of learning new things and finally got around to watching your 2021 4 hour workflow video. Due to this, I've refactored my entire vault, but I have a few questions that maybe you could explain:

1. Why do you use evergreen note type tags, and not just colored squares? There are 4 tags, either way, and adding the evergreen tags seems redundant when the squares seem to work fine.
2. What's the logic behind using red squares in the tags frontmatter, vs the status frontmatter? I get you can search by tags sometimes more easily, but with dataview you can search by status as well. And it seems like sometimes it's in the tags field, and sometimes in the status field? I'd like to keep it in one place and that's it personally so I don't have to remember extra rules. 
3. What is the purpose of aliases and cssclass in frontmatter? I've been just leaving them blank for now as I haven't needed to use them it seems. 
4. why do you sometimes have a new section below the frontmatter with seemingly extra fields? Maybe I'm wrong, but I think I remember seeing a spot for tags once or twice. Why not just keep all this in the frontmatter whenever possible? I know you can't add wikilinks in frontmatter, and im sure there are other situations for it. But could you explain your reasoning?

Thank you and I hope to hear from you soon!

**Comment by tallguyjenks** [comment, marked as answer]
1. I like them, and i use the square elsewhere for other things so I don't want them to blend in my mind
2. status is for the supercharged links plugin that is all, and doesn't show a child tag in the nested tags pane
3. check out the help vault 
4. I explain this in my 4 hour video, I add wiki links there to [[tags]] not #tags and other pieces of information relevant to my workflow

**Reply by KjellConnelly** [reply]
Thanks for the reply :)

1. Is that because evergreen notes are more organic ideas you create on your own - very personal. Whereas 'input' types are more someone else's ideas that you're recapping, etc? Or is it just the visual you like?
2. I get it now. Sometimes you want a note's status shown in the nested tags. Sometimes you want it inline in your wikilinks. And sometimes you want both. I'm not completely sure if there are cases where I'd not want both, and if that were the case, it seems redundant. But maybe there is a way to have the one spot for status instead of including it on two lines. But maybe that's just more work than it's worth due to needing a custom solution.
3. Alright, I got aliases working well. cssclass, i'll be leaving empty for now. Maybe someday I'll care enough.
4. I'm relistening to the video to/from work, so I'll let you know if I am still confused on this one after I learn some more.

On a side note, and I'm not sure if you already have a solution to this, but I saw you using `textsniper` and copy screenshots to clipboard on macOS. I'm using the same thing, and since I'm syncing my vault to GitHub, I ended up having many screenshots in much higher quality than I wanted and taking up too much space. So I wrote a little bash script to minimize images that are imported this way. It seems that whenever I paste new images into Obsidian, they're pasted with filenames always like `Pasted image 789316396931.png` or something like that. My script looks for all files in the `Resources` directory, and uses `ImageMagick` to turn them into smaller jpgs. Typically turns files that are 2-3mb into 100-200kb images, which is nice for me usage. Then it uses `sed` to fix filenames in all `.md` files (except my code directory). If you want to use this script, go for it. If not, I just thought you might find it interesting.

```bash
#!/bin/sh

# Find all Pasted image .png files in Resources directory
cd Resources
mkdir -p temp/output
find . -name "*Pasted image*.png" -exec mv {} temp/ \;
cd temp

# Minimize png images to jpg format
mogrify -strip -quality 80 -path output -format jpg -resize 1080x720\> *.png
mv output/* ../
rmdir output
cd ../

# Move png files to system trash in a timestamped directory
now=$(date -u +"%Y-%m-%dT%H-%M-%SZ")
mv temp originalPngs-"$now"
mv originalPngs-"$now" ~/.Trash

# find all md files, not in the 'Z/code' directory. Fix references from png to jpg files.
find . -name "*.md" -not -path './Z/code/*' -exec sed -i "" "s/Pasted image \([^)]*\).png/Pasted image \1.jpg/" {} \;
```

**Reply by KjellConnelly** [reply]
Hi again! I realized that my script has the same `sed` problem you talked about and was setting all `ctime` data to the date I ran the script. So I swapped out that last `find . -name...` in the script above and run a `nodejs` script (I'm a much better node programmer than bash). Just requires you `npm init` and `npm i replace-in-file` to use. Feel free to use this script instead of `sed` if you'd like.

```javascript
const replace = require('replace-in-file')

function pastedImagePngsToJpgs() {
  const options = {
    countMatches:true,
    files: ['**/*.md'],
    ignore: ['Z/code/*'],
    from: /Pasted image (?<number>.*?).png/g,
    to: 'Pasted image $1.jpg'
  }
  const results = replace.sync(options)
  const changedFiles = results.filter(result => result.hasChanged)
  console.log(changedFiles)
}

// call it here, or do something with pastedImagePngsToJpgs()
```

**Reply by tallguyjenks** [reply]
1. the visual, the ever green tags have nothing to do with my inputs 

thanks for the script, I don't use textsniper for images only OCR on text and for the TTS :)
